src/circuit/auto-relay.js

```diff
@@ -129,28 +128,19 @@ class AutoRelay {
       return
     }
 
-    // Create relay listen addr
-    let listenAddr, remoteMultiaddr, remoteAddrs
-
-    try {
-      // Get peer known addresses and sort them per public addresses first
-      remoteAddrs = this._peerStore.addressBook.get(connection.remotePeer)
-      // TODO: This sort should be customizable in the config (dialer addr sort)
-      remoteAddrs.sort(multiaddrsCompareFunction)
-
-      remoteMultiaddr = remoteAddrs.find(a => a.isCertified).multiaddr // Get first announced address certified
-      // TODO: HOP Relays should avoid advertising private addresses!
-    } catch (_) {
-      log.error(`${id} does not have announced certified multiaddrs`)
-
-      // Attempt first if existing
-      if (!remoteAddrs || !remoteAddrs.length) {
-        return
-      }
+    // Get peer known addresses and sort them per public addresses first
+    // Sorting addresses to public first
+    const remoteAddrs = this._addressSorter(
+      this._peerStore.addressBook.get(connection.remotePeer) || []
```

Comment:
Use `getMultiaddrsForPeer`? This should also avoid the `if` check below for the `p2p` protoName.